File: examples/human/email/dotnet/EmailAgent.cs
using System.Text.Json.Nodes;
using Axme.Sdk;
using AxmeExamples;

namespace AxmeExamples.Human;

public class EmailAgent
{
    public static async Task Main(string[] args)
    {
        var baseUrl = SseHelper.Env("AXME_BASE_URL", "https://api.cloud.axme.ai");
        var apiKey = SseHelper.Env("AXME_API_KEY");
        var addr = SseHelper.Env("AXME_AGENT_ADDRESS", "budget-envelope-validator");
        if (string.IsNullOrEmpty(apiKey)) { Console.Error.WriteLine("AXME_API_KEY is required"); return; }

        var client = new AxmeClient(new AxmeClientConfig { ApiKey = apiKey, BaseUrl = baseUrl });
        Console.WriteLine($"budget-envelope-validator starting  address={addr}");
        var envelopes = new Dictionary<string, double> { ["cloud_infrastructure"] = 50000, ["software_licenses"] = 20000, ["contractor_services"] = 40000, ["hardware"] = 15000 };
        var since = 0;

        while (true)
        {
            try
            {
                foreach (var d in await SseHelper.PollAgentStreamAsync(baseUrl, apiKey, addr, since))
                {
                    var id = SseHelper.Str(d, "intent_id");
                    if (d["seq"]?.GetValue<int>() is int seq) since = Math.Max(since, seq);
                    if (string.IsNullOrEmpty(id)) continue;
                    var intent = SseHelper.UnwrapIntent(await client.GetIntentAsync(id));
                    if (!SseHelper.IsActionable(intent)) continue;
                    var p = SseHelper.EffectivePayload(intent);

                    var amount = p["amount"]?.GetValue<double>() ?? 0;
                    var category = SseHelper.Str(p, "category");
                    var currency = SseHelper.Str(p, "currency", "USD");

                    JsonObject res;
                    if (!envelopes.TryGetValue(category, out var envelope))
                        res = new JsonObject { ["action"] = "fail", ["within_envelope"] = false, ["reason"] = "unknown category" };
                    else if (amount > envelope)
                        res = new JsonObject { ["action"] = "fail", ["within_envelope"] = false, ["reason"] = "exceeds envelope" };
                    else
                    {
                        var headroom = Math.Round((envelope - amount) / envelope * 1000) / 10;
                        res = new JsonObject { ["action"] = "complete", ["within_envelope"] = true,
                            ["reason"] = $"{currency} {amount:N0} within {category} ({headroom}% headroom)",
                            ["envelope"] = envelope, ["headroom_pct"] = headroom };
                    }
                    await client.ResumeIntentAsync(id, res, new RequestOptions { OwnerAgent = addr });
                    Console.WriteLine($"resumed {id} (within_envelope={res["within_envelope"]})");
                }
            }
            catch { await Task.Delay(2000); }
        }
    }
}

File: examples/human/email/java/EmailAgent.java
package ai.axme.examples.human;

import ai.axme.examples.SseHelper;
import dev.axme.sdk.*;
import java.util.*;

/** Budget Envelope Validator — human/email example (Java). */
public class EmailAgent {
    static final String BASE = SseHelper.env("AXME_BASE_URL", "https://api.cloud.axme.ai");
    static final String KEY = System.getenv("AXME_API_KEY");
    static final String ADDR = SseHelper.env("AXME_AGENT_ADDRESS", "budget-envelope-validator");
    static final Map<String, Double> ENVELOPES = Map.of(
            "cloud_infrastructure", 50000.0, "software_licenses", 20000.0,
            "contractor_services", 40000.0, "hardware", 15000.0);

    public static void main(String[] args) throws Exception {
        if (KEY == null || KEY.isEmpty()) { System.err.println("AXME_API_KEY is required"); System.exit(1); }
        AxmeClient client = new AxmeClient(new AxmeClientConfig(BASE, KEY));
        System.out.printf("budget-envelope-validator starting  address=%s%n", ADDR);
        int since = 0;
        while (true) {
            try {
                for (Map<String, Object> d : SseHelper.pollAgentStream(BASE, KEY, ADDR, since, 15)) {
                    String id = SseHelper.str(d, "intent_id"); Number seq = (Number) d.get("seq");
                    if (seq != null) since = Math.max(since, seq.intValue()); if (id.isEmpty()) continue;
                    Map<String, Object> intent = SseHelper.unwrapIntent(client.getIntent(id, RequestOptions.none()));
                    if (!SseHelper.isActionable(intent)) continue;
                    Map<String, Object> p = SseHelper.effectivePayload(intent);

                    double amount = p.get("amount") instanceof Number n ? n.doubleValue() : 0;
                    String category = SseHelper.str(p, "category"), currency = SseHelper.str(p, "currency", "USD");
                    Double envelope = ENVELOPES.get(category);

                    Map<String, Object> res = new HashMap<>();
                    if (envelope == null) { res.put("action", "fail"); res.put("within_envelope", false); res.put("reason", "unknown category"); }
                    else if (amount > envelope) { res.put("action", "fail"); res.put("within_envelope", false); res.put("reason", "exceeds envelope"); }
                    else {
                        double headroom = Math.round((envelope - amount) / envelope * 1000.0) / 10.0;
                        res.put("action", "complete"); res.put("within_envelope", true);
                        res.put("reason", String.format("%s %.0f within %s envelope (%.1f%% headroom)", currency, amount, category, headroom));
                        res.put("envelope", envelope); res.put("headroom_pct", headroom);
                    }
                    client.resumeIntent(id, res, SseHelper.withOwner(ADDR));
                    System.out.printf("resumed %s (within_envelope=%s)%n", id, res.get("within_envelope"));
                }
            } catch (Exception e) { Thread.sleep(2000); }
        }
    }
}
